### ISSUE TITLE
[prefer-output-readonly] stops working

### ISSUE THREAD

**Issue by de-don** [angular-eslint/angular-eslint#1787]
The rule `@angular-eslint/prefer-output-readonly` stops working after updating to angular 16.

```JSON
{
  "files": ["*.ts"],
  "extends": [
    "plugin:@nx/typescript",
    "plugin:@nx/angular",
    "plugin:@angular-eslint/template/process-inline-templates"
  ],
  "rules": {
    //...
    "@angular-eslint/prefer-output-readonly": "error",
    //...
  }
}
```

```TS
  @Output()
  public emitter1 = new EventEmitter<void>();
  
  @Output() public emitter2 = new EventEmitter<void>();
```

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@angular-eslint/eslint-plugin`    | `16.3.1` |
| `@typescript-eslint/parser`        | `6.21.0` |
| `ESLint`                           | `8.48.0` |
| `node`                             | `18.18.2` |

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest supported version of the packages and checked my `ng version` output per the instructions given here.

**Comment by de-don** [comment]
Hi, I just checked. It still reproduces. Part of my config:
```
{
  "root": true,
  "plugins": ["@nx", "deprecation", "import", "jest", "jsdoc", "@angular-eslint"],
  "overrides": [
    // ...
    {
      "files": ["*.ts"],
      "extends": [
        "plugin:@nx/typescript",
        "plugin:@nx/angular",
        "plugin:@angular-eslint/template/process-inline-templates"
      ],
      "rules": {
        "@angular-eslint/no-outputs-metadata-property": "off",
        "@angular-eslint/no-host-metadata-property": "error", // this rule works correctly
        "@angular-eslint/prefer-output-readonly": "error", // this rule doesn't work
      }
    }
  ]
```

**Comment by de-don** [comment]
My research has yielded the following results:

If I update the source code of the rule before that, everything will work as expected (version 16.3.1):

```js
    create(context) {
        return {
            [`${utils_1.Selectors.OUTPUT_DECORATOR}`](val) {  // <----- Here I removed first part with readonly
                if (val.parent.readonky) {  // <----- Here I have added readonly check
                    return;
                }
  

                const key = val.parent.key;
                context.report({
                    node: key,
                    messageId: 'preferOutputReadonly',
                    suggest: [
                        {
                            messageId: 'suggestAddReadonlyModifier',
                            fix: (fixer) => fixer.insertTextBefore(key, 'readonly '),
                        },
                    ],
                });
            },
        };
    },

**Comment by de-don** [comment]
I found the root case.

Need to update `PropertyDefinition:not([readonly])` to this `PropertyDefinition:not([readonly=true])` and this will solve the problem. Looks like it's already fixed in newer version, but in 16.3.x it's not fixed :(

**Comment by JamesHenry** [comment]
I'm afraid as unpaid volunteers we don't have scope to support multiple outdated major versions, but it seems this should make for a pretty trivial patch to apply to your node_modules whilst stuck on Angular 16 via:

- `pnpm patch`
- `yarn patch`
- https://www.npmjs.com/package/patch-package (with any package manager)
